com/bl/ttt/tictactoe.py

```diff
@@ -49,15 +49,14 @@ def get_previous_turn_player(self):
         else:
             return self.player1
 
-    def get_player_move_log(self):
-        return self.player_move_log
+    def get_player_move_log(self, player_id='both'):
+        if player_id == 'both':
+            return f"X played cells {self.player_move_log.get('X')} and O played cells {self.player_move_log.get('O')}."
+        else:
+            return f'{player_id} played cells {self.player_move_log.get(player_id)}.'
```

Comment:
Don't make assumptions about what the client wants to do with results.
Let the client decide what it wants to do with the results so return just the list of moves.

Reply:
Also - return a copy of the internal list so that there is no danger of the client changing the moves made or making the move log inconsistent with the results.

Reply:
OK re not making assumptions about what the client wants to do with results.

OK re  returning a copy of the internal list to prevent log mutation.

Reply:
get_player_move_log now returns ordered lists for individual players: eg. [2, 4, 7]

Reply:
get_player_move_log now returns a list for both players together in order: say player_move_log = {'X': [2, 4, 7], 'O': [1, 3, 8]} then [2, 1, 4, 3, 7, 8] is returned.